src/main/java/org/apereo/openlrs/controllers/caliper/CaliperController.java

```diff
@@ -86,28 +131,86 @@ public JsonNode getByIdHandler (
      *
      * @param activity
      * @param actor
-     * @param limit
      * @param since
      * @param until
+     * @param pageParam Number of the requested page
+     * @param limitParam Number of results per page
+     * @param numberParam Alias for pageParam, corresponds with "number" response attribute (pageParam has priority)
+     * @param sizeParam Alias for limitParam, corresponds with "size" response attribute (limitParam has priority)
      * @return JSON string of the statement objects matching the specified filter
      */
-    @RequestMapping(value = {"", "/"}, method = RequestMethod.GET,
-            produces = "application/json;charset=utf-8", params = "!statementId")
-    public List<JsonNode> getByFiltersHandler (
+    @RequestMapping(
+            method = RequestMethod.GET,
+            value = {"", "/"},
+            params = {"!statementId",},
+            produces = "application/json;charset=utf-8"
+    )
+    @ResponseBody
+    public Page<JsonNode> getByFiltersHandler(
             @RequestParam(value = "actor", required = false) String actor,
             @RequestParam(value = "activity", required = false) String activity,
             @RequestParam(value = "since", required = false) String since,
             @RequestParam(value = "until", required = false) String until,
-            @RequestParam(value = "limit", required = false) String limit
-    ) throws InvalidRequestException {
-        try {
-            Map<String, String> filterMap = null;
-            filterMap = StatementUtils.createStatementFilterMap(actor, activity, since, until, limit);
-            return caliperService.getJsonNodes(filterMap);
-        } catch (Exception e) {
-            logger.error(e.getMessage(), e);
-            throw new InvalidRequestException(e.getMessage(), e);
+            @RequestParam(value = "page", required = false) String pageParam,
+            @RequestParam(value = "limit", required = false) String limitParam,
+            @RequestParam(value = "number", required = false) String numberParam,
+            @RequestParam(value = "size", required = false) String sizeParam
+    ) throws NumberFormatException {
+        Map<String, String> filterMap =
+                StatementUtils.createStatementFilterMap(actor, activity, since, until, limitParam);
+        Integer page = this.getPageDefault();
+        String pageParamName = "page";
+        Integer limit = this.getLimitDefault();
+        String limitParamName = "limit";
+        Integer limitMaximum = this.getLimitMaximum();
+
+        if ((pageParam == null) && (numberParam != null)) {
+            pageParam = numberParam;
+            pageParamName = "number";
         }
+
```

Comment:
Reformat the code for getting pageParam and limitParam be reformatted and use a common method?